src/queryParameters/extractOpportunityIdParameter.ts

```diff
@@ -0,0 +1,19 @@
+import apiSpecification from '../openapi.json';
+import { SimpleInputValidationError } from '../errors';
+import type { Request } from 'express';
+
+export const extractOpportunityIdParameter = (
+  request: Request,
+) => {
+  if (request.query.opportunityId !== undefined
+    && request.query.opportunityId !== '') {
+    const id = Number(request.query.opportunityId);
+    if (!Number.isSafeInteger(id)) {
+      throw new SimpleInputValidationError(`opportunityId must be an integer, got '${id}'.`);
```

Comment:
In this domain, I think we should be narrower: we want a positive integer, as zero or a negative number will never be a valid opportunity ID. (Right?)

To that end, I think we might want to continue to use AJV here, rather than introduce a new `SimpleInputValidationError` type, with a schema like `{ type: 'integer', minimum: 1 }`.

Reply:
Fair enough, yes.

Reply:
We have a pattern for this kind of check which uses ajv in the pagination parameter tooling